cmake/compiler_settings.cmake

```diff
@@ -70,6 +70,16 @@ else ()
 endif ()
 
 if (CMAKE_CXX_COMPILER_ID MATCHES GNU OR CMAKE_CXX_COMPILER_ID MATCHES Clang)
+  # Enforce -O0 in debug mode.
+  string(TOUPPER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE_UPPER)
+  if (CMAKE_BUILD_TYPE_UPPER STREQUAL "DEBUG")
+    # For gcc the -g option alredy includes -O0.
```

Comment:
Is this change related to the changing of the default `CMAKE_BUILD_TYPE` in the PR description? It's seems like a good change to enforce consistency, but the PR should probably be updated to explain the impact of this fix instead.

@anakrish I wasn't aware that we default to `-O0` for debug builds; for my understanding, is there a reason for that instead of using `-Og`, or should we be using that instead for both gcc & clang?

Reply:
@CodeMonkeyLeet Thanks for your explanation! Anand has answered in another pr, just because that pr is not merge yet so the change also shows up here.
The gcc -g already include -O0 by default, but clang does not.
Anand also mentioned if -Og is a better choice. For clang -Og is still under development. for gcc -Og has some optimization that some symbols are removed and backtrace test can not pass.

Reply:
@nonpolarity Thanks for the explanation, wasn't aware for the clang differences in behavior. Can we just drop commit dd9bfc13e054e49afc57ab7a741e8cffbcbf172f and just take this one then?